system/docker/node/Dockerfile

```diff
@@ -1,76 +1,81 @@
 FROM solita/ubuntu-systemd
 
+ARG uid=1000
+ARG user=indy
+
 # Install environment
 RUN apt-get update -y && apt-get install -y \
-        git \
-        wget \
-        python3.5 \
-        python3-pip \
-        python-setuptools \
-        python3-nacl \
-        apt-transport-https \
-        ca-certificates \
-        ssh \
-        vim \
-        mc \
-        stress \
-	&& rm -rf /var/lib/apt/lists/*
+    git \
+    wget \
+    python3-pip \
+    python3-nacl \
+    python-setuptools \
+    python3-nacl \
+    apt-transport-https \
+    ca-certificates \
+    ssh \
+    vim \
+    mc \
+    stress \
+    dbus \
+    && rm -rf /var/lib/apt/lists/*
 
+# pip<10.0.0 is important. pip==10.0.0 doesn't work.
 RUN pip3 install -U \
-        'pip<10.0.0' \
-        setuptools
+    'pip<10.0.0' \
+    setuptools
+
+# sovrin repo init
+ARG NODE_REPO_COMPONENT
+RUN : ${NODE_REPO_COMPONENT:?"NODE_REPO_COMPONENT must be provided"}
+ENV NODE_REPO_COMPONENT=${NODE_REPO_COMPONENT}
 
-ARG INDY_NODE_REPO_COMPONENT
+RUN apt-key adv --keyserver keyserver.ubuntu.com --recv-keys CE7709D068DB5E88 \
+    && echo "deb https://repo.sovrin.org/deb xenial ${NODE_REPO_COMPONENT}" >> /etc/apt/sources.list \
+    && apt-get update -y
+
+# indy depedencies
 ARG LIBINDY_CRYPTO_VERSION
+RUN : ${LIBINDY_CRYPTO_VERSION:?"LIBINDY_CRYPTO_VERSION must be provided"}
```

Comment:
provides greater control and better error feedback when using env vars to control the test behaviour.